packages/api/src/codegen/factory.ts

```diff
@@ -4,9 +4,14 @@ import type Oas from 'oas';
 
 import TSGenerator from './languages/typescript/index.js';
 
-export enum SupportedLanguages {
-  JS = 'js',
-}
+export const SupportedLanguages = {
+  JS: 'js',
+} as const;
+
+type ObjectValues<T> = T[keyof T];
+
+export type SupportedLanguageType = ObjectValues<typeof SupportedLanguages>;
```

Comment:
I don't think we have any TS types that end in `Type` anywhere else here so may as well be consistent.

```suggestion
export type SupportedLanguage = ObjectValues<typeof SupportedLanguages>;
```